keps/sig-scheduling/4816-dra-prioritized-list/README.md

```diff
@@ -594,9 +614,21 @@ type DeviceRequest struct {
     // +oneOf=deviceRequestType
     DeviceClassName string
 
-    // FirstAvailable contains subrequests, exactly one of which must be satisfied
-    // in order to satisfy this request. This field may only be set in the
-    // entries of DeviceClaim.Requests.
+    // FirstAvailable contains subrequests, at least one of which must be
+    // satisfied in order to satisfy this request. This field may only be
+    // set in the entries of DeviceClaim.Requests.
+    // 
+    // The scheduler will attempt to satisfy the list of subrequests in
+      // order, and it will select the first one that satisfies the
+    // requirements of the request. So if there are two entries in the
+    // list, the schduler will only check the second one if it determines
+    // that the first one can not be used.
+    //
+    // DRA does not yet implement scoring, so the scheduler will
+    // select the first set of devices that satisfies all the
```

Comment:
This seems to describe the logic locally to a node. But it doesn't explain how a node is chosen.

If a node A satisfies option 2, and node B satisfies option 3, there is nothing in this design that prevents the scheduler to choose node B.

Reply:
Yeah, I see what you are referring to. I had to look into the behavior here, as the scheduler plugin for DRA does not implement the `ScorePlugin` interface. From what I can tell, the behavior in this situation is that the scheduler will choose one of the nodes that satisfy the requirements.

So for the example, you are correct that node B might be chosen in this situation, even if it seems like node A would be the better alternative. I've updated the language to reflect this.

@pohly to confirm the behavior here.

Reply:
Yes, that's correct. Without scoring, this is a node-local decision. When nodes are homogenous and there are plenty of resources, this is not super useful. It's useful for heterogeneous nodes (those with different types of devices, which will be the case when used along with the *partitionable* KEP), and it is useful when resources are scarce. In other words, if you stock out on H100s, this will let you use a node with A100s.